cornell-dti/course-plan
# Fall 2022 Pre-Enroll Release

# Summary 

This pull request introduces the Tools page, featuring personalized advisor information, a progress tracker, a PDF schedule export option, and a collection of useful links! It also updates the course data in preparation for Spring 2023 pre-enroll and improves the development workflow.

## Changes

### Tools Page
- [x] Tools Page, Advisors Card, and Advisors Backend #690 (@noschiff)
- [x] Improve Tool Cards and Tools Page #719 (@noschiff @zachary0kent)
- [x] Tools Page Formatting & Useful Links #726 (@noschiff)
- [x] Touch Up Useful Links Tool Card #740 (@noschiff @benjamin-shen)
- [x] Add Advisors Data (#734 (@noschiff @zachary0kent)
- [x] Create Export Schedule Tool Card #748 (@noschiff)
- [x] Relocate Requirement Progress Calculation #750 (@noschiff)
- [x] Create Progress Tracker Tool Card #751 (@noschiff @zachary0kent @andxu282)
- [x] Function to generate user's course PDF #730 (@vaishnavi17 @zachary0kent @benjamin-shen @noschiff)
- [x] Remove tools feature flag #746 (@zachary0kent)
- [x] Fix Navigation Bug #757 (@zachary0kent)
- [x] Small PDF generation bug fixes #758 (@vaishnavi17)
- [x] Quick Touch Up to Tools Page #760 (@noschiff)

### Workflow and Utility
- [x] Fix failing requirements sidebar Cypress test #697 (@willespencer)
- [x] Copy Data Between Users Across Dev & Prod #711 (@noschiff)
- [x] Fix Script Copy Data Between Users #715 (@noschiff)
- [x] refactor global firestore data files #717 (@benjamin-shen)
- [x] update contributors #718 (@noschiff)
- [x] remove firebase functions #721 (@benjamin-shen)
- [x] Clean Data in Course Generator Script #720 (@noschiff)
- [x] Update Courses for Winter and Spring 2023 #727 (@noschiff)
- [x] Fix requirements json after #727 #729 (@benjamin-shen)
- [x] Update scripts #724 (@benjamin-shen)
- [x] Move docs to wiki #731 (@benjamin-shen)
- [x] Clean up workflows, add update-courses workflow #728 (@benjamin-shen)
- [x] Write script outputs to ignored directory #735 (@benjamin-shen)
- [x] Update Copy User Data Script Documentation (#739 (@noschiff)
- [x] Update update-courses workflow #736 (@benjamin-shen)
- [x] Remove semester type from database #595 (@benjamin-shen)
- [x] Fix linting for enums #752 (@zachary0kent)
- [x] Update courseplan email #755 (@benjamin-shen)
- [x] Update Courses (FA22-Nov1) #756 (@benjamin-shen @noschiff)
- [x] Reorganize tools in codebase #759 (@benjamin-shen)

### Requirements
- [x] Implement Alternate Options for Engineering Physics Major #687 (@noschiff)
- [x] Add psych major #709 (@benjamin-shen @kkkehui)
- [x] rename human policy to health policy #733 (@benjamin-shen)

### Gatekept Features

- [x] Two-graph requirement header progress colors and warnings #622 (@willespencer @SamChou19815)
- [x] Populate conflict modal with requirements #669 (@willespencer @SamChou19815)
- [x] Setup backend infra for CASE exams #641 (@benjamin-shen @willespencer)
- [x] Add transfer credits and save confirmation modal (#691)  (@jerrry1123 @willespencer)
- [x] Added new profile container with the first page of settings #675 (@jerrry1123)
- [x] capitalize 'get' in profile container #732 (@benjamin-shen)
- [x] Profile Page Feature Flag #743 (@zachary0kent)

# Test Plan 

Confirm everything works as expected on all pages of CoursePlan site, especially on the Tools page and navigating between pages, with no console errors. 

For example:
- Navigate between the plans page, tools page, and profile pop-up and ensure you get where you expect to be!
- Collapse and expand the Tool Cards and resize the screen for the Tools Page. You can refresh to ensure the collapsed state persists!
- Add some majors and minors and look at the advisors!
- Put some courses into your schedule and download the PDF!
- Ensure progress tracker displays information for all colleges/majors/minors/grad programs and the bar represents the correct progress.
- Visit the useful links!
- Check that a new course such as AEP 3100 can be added and that the bottom bar shows up with expected information for it.

<img alt="preview" src="https://user-images.githubusercontent.com/47431797/199834740-36744d05-8a49-4d01-9368-d3ebe323ded2.png">

# Notes 
Do not squash and merge.
